core-tests/shared/src/test/scala/zio/FunctionIOSpec.scala

```diff
@@ -1,202 +1,143 @@
-               succeed("is negative")
-             ).run(1)
-      } yield (v1 must_=== "is negative") and (v2 must_=== "is positive")
-    )
-
-  def e15a =
-    unsafeRun(
-      for {
-        v <- whileDo[Nothing, Int](fromFunction[Int, Boolean](_ < 10))(fromFunction[Int, Int](_ + 1)).run(1)
-      } yield v must_=== 10
-    )
-
-  def e15b =
-    unsafeRun(
-      for {
-        v <- whileDo[Nothing, Int](fromFunctionM[Nothing, Int, Boolean](a => IO.succeed[Boolean](a < 10)))(
-              fromFunctionM[Nothing, Int, Int](a => IO.effectTotal[Int](a + 1))
-            ).run(1)
-      } yield v must_=== 10
-    )
-
-  def e16 =
-    unsafeRun(
-      for {
-        v <- _1[Nothing, Int, String].run((1, "hi"))
-      } yield v must_=== 1
-    )
-
-  def e17 =
-    unsafeRun(
-      for {
-        v <- _2[Nothing, Int, String].run((2, "hola"))
-      } yield v must_=== "hola"
-    )
-
-  def e18a =
-    unsafeRun(
-      for {
-        a <- fail[String]("error").run(1).either
-      } yield a must_=== Left("error")
-    )
-  def e18b =
-    unsafeRun(
-      for {
-        a <- effect[String, Int, Int] { case _: Throwable => "error" }(_ => throw new Exception).run(9).either
-      } yield a must_=== Left("error")
-    )
-
-  def e19 =
-    unsafeRun(
-      for {
-        a <- effect[String, Int, Int] { case _: Throwable => "error" }(_ => throw new Exception).run(9).ignore
-      } yield a must be_==(())
-    )
+  val thrower = effect[String, Int, Int] { case _: Throwable => "error" }(
+    _ => throw new Exception
+  )
 }
+object FunctionIOSpec
+    extends ZIOBaseSpec(
+      suite("FunctionIOSpec")(
+        suite("Check if the functions in `FunctionIO` work correctly")(
+          testM("`fromFunction` lifts from A => B into effectful function") {
+            assertM(add1.run(4), equalTo(5))
+          },
+          testM("`identity` returns the identity of the input without modification") {
+            assertM(identity[Int].run(1), equalTo(1))
+          },
+          testM("`>>>` is a symbolic operator of `andThen`which does a Backwards composition of effectful functions") {
+            assertM((add1 >>> mul2).run(6), equalTo(14))
+          },
+          testM("`<<<` is a symbolic operator of `compose` which compose two effectful functions") {
+            assertM((add1 <<< mul2).run(6), equalTo(13))
+          },
+          testM("`zipWith` zips the output of two effectful functions") {
+            assertM(add1.zipWith(mul2)(_ -> _).run(6), equalTo(7 -> 12))
+          },
+          testM("`&&&` zips the output of two effectful functions and returns a tuple of their result") {
+            assertM((add1 &&& mul2).run(6), equalTo(7 -> 12))
+          },
+          testM("`|||` computes two effectful functions left and right from from an Either input") {
+            for {
+              l1 <- (add1 ||| mul2).run(Left(25))
+              r1 <- (add1 ||| mul2).run(Right(25))
+            } yield (assert(l1, equalTo(26))
```

Comment:
You can get rid of the outer parentheses here and in other tests where you are composing assertions.

Reply:
Just put the `&&` at the end of each line.